matrix/common/s3_zarr_store.py

```diff
@@ -102,15 +103,31 @@ def _write_row_data_to_results_chunk(self, chunk_idx: int, input_bounds: tuple,
             # Reading and writing zarr chunks is pretty straightforward, you
             # just pass it through the compression and cast it to a numpy array
             with Lock(full_dest_key):
-                try:
-                    arr = numpy.frombuffer(
-                        ZARR_OUTPUT_CONFIG['compressor'].decode(
-                            self.s3_file_system.open(full_dest_key, 'rb').read()),
-                        dtype=dtype).reshape(chunk_shape, order=ZARR_OUTPUT_CONFIG['order'])
-                except FileNotFoundError:
-                    arr = numpy.zeros(shape=chunk_shape,
-                                      dtype=dtype)
-                    print("Created new array")
+                # This is a graceless workaround for the issue identified here:
+                # https://github.com/pangeo-data/pangeo/issues/196
+                # Sometimes blosc.decode raises a RuntimeError that goes away on retry
+                # TODO: factor out the retry logic
```

Comment:
I'm thinking this would look like a decorator that we could apply on a function that extracts the decomp logic? Fine for now